src/server/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp

```diff
@@ -360,14 +358,16 @@ class boss_malchezaar : public CreatureScript
             {
                 if (SWPainTimer <= diff)
                 {
-                    Unit* target = nullptr;
-                    if (phase == 1)
-                        target = me->GetVictim();                  // Target the Tank
-                    else                                          // anyone but the tank
-                        target = SelectTarget(SELECT_TARGET_RANDOM, 1, 100, true);
+
+                    // if phase == 1 target the tank, otherwise anyone but the tank
+                    Unit* target = phase == 1
```

Comment:
all this unit target can be defined as `const` I think

Reply:
@Helias I guess so, but I just did not want to make too many chances at once in so many files.

During these refactoring PRs I just do the minimal/obvious changes to fix silly mistakes or sometimes prevent crashes. I know changing the type it's often quite trivial, but sometimes it can lead to undesired side effects so I did not bother too much checking the types